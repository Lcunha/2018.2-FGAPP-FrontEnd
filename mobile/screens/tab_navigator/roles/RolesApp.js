import React, { Component } from "react";
import { 
    View,
    Text,
    StyleSheet
} from "react-native";

import HomeScreen from './screens/HomeScreen'
import CadastroEventos1 from './screens/TabNavigator/cadastroEventos/CadastroEventos1'
import CadastroEventos2 from './screens/TabNavigator/cadastroEventos/CadastroEventos2'
import Feed from './screens/TabNavigator/Feed'
import {StackNavigator} from 'react-navigation'
import RolesTabHandler from './RolesTabHandler'
import * as firebase from 'firebase'


export default class RolesApp extends Component {

    componentWillMount() {
        var config = {
            apiKey: "AIzaSyAHz3Y9yDmb4Pk5rvp6vdgTZscweubgTC8",
            authDomain: "react-native-roles.firebaseapp.com",
            databaseURL: "https://react-native-roles.firebaseio.com",
            projectId: "react-native-roles",
            storageBucket: "react-native-roles.appspot.com",
            messagingSenderId: "348737140329"
        };
        firebase.initializeApp(config);        
    }

    render() {
        return (
            <RolesStackNavigator />
        );
    }
}

const RolesStackNavigator = new StackNavigator({
    RolesTabHandler:{
        screen:RolesTabHandler,
        navigationOptions: ({ navigation }) => ({
          header: null,
        }),
    },
    HomeScreen:{
        screen:HomeScreen,
        navigationOptions: {
            header: null
        }
    },
    CadastroEventos1:{
        screen:CadastroEventos1,
        navigationOptions:{
            header:null
        },
    },
    CadastroEventos2:{
        screen:CadastroEventos2,
        navigationOptions:{
            header:null
        },
    },
    Feed: {
        screen: Feed,
        navigationOptions:{
            header:null
        },
    },
    
})

//Cor limegreen principal do app
// #32CD32