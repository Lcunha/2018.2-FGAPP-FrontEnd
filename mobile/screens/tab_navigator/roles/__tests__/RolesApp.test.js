import React from 'react';
import RolesApp from '../RolesApp'
import Adapter from 'enzyme-adapter-react-16';
import {shallow} from 'enzyme';
import Enzyme from 'enzyme';
import renderer from 'react-test-renderer';

Enzyme.configure({adapter: new Adapter()});

it('renders correctly', () => {
  const tree = renderer.create(<RolesApp />).toJSON();
  expect(tree).toMatchSnapshot();
});
