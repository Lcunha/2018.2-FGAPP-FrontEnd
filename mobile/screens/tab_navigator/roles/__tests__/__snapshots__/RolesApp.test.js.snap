// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  collapsable={false}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Object {
          "flex": 1,
          "flexDirection": "column-reverse",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          collapsable={undefined}
          pointerEvents="auto"
          style={
            Object {
              "backgroundColor": "#E9E9EF",
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "shadowColor": "black",
              "shadowOffset": Object {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0.4,
              "shadowRadius": 10,
              "top": 0,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
                Object {
                  "translateY": 0,
                },
              ],
            }
          }
        >
          <View
            loaded={
              Array [
                0,
              ]
            }
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flex": 1,
                  "overflow": "hidden",
                },
                Object {
                  "flex": 1,
                },
              ]
            }
          >
            <View
              collapsable={undefined}
              onMoveShouldSetResponder={[Function]}
              onMoveShouldSetResponderCapture={[Function]}
              onResponderEnd={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderReject={[Function]}
              onResponderRelease={[Function]}
              onResponderStart={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              onStartShouldSetResponderCapture={[Function]}
              style={
                Object {
                  "alignItems": "stretch",
                  "flex": 1,
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
                testID={undefined}
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "overflow": "hidden",
                    }
                  }
                >
                  <View
                    onLayout={[Function]}
                    style={
                      Array [
                        Object {
                          "flex": 1,
                        },
                        undefined,
                      ]
                    }
                  >
                    <View
                      onMoveShouldSetResponder={[Function]}
                      onMoveShouldSetResponderCapture={[Function]}
                      onResponderEnd={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderReject={[Function]}
                      onResponderRelease={[Function]}
                      onResponderStart={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      onStartShouldSetResponderCapture={[Function]}
                      style={
                        Object {
                          "flex": 1,
                          "flexDirection": "column-reverse",
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "flex": 1,
                          }
                        }
                      >
                        <View
                          collapsable={undefined}
                          pointerEvents="auto"
                          style={
                            Object {
                              "backgroundColor": "#E9E9EF",
                              "bottom": 0,
                              "left": 0,
                              "opacity": 1,
                              "position": "absolute",
                              "right": 0,
                              "shadowColor": "black",
                              "shadowOffset": Object {
                                "height": 0,
                                "width": 0,
                              },
                              "shadowOpacity": 0.4,
                              "shadowRadius": 10,
                              "top": 0,
                              "transform": Array [
                                Object {
                                  "translateX": 0,
                                },
                                Object {
                                  "translateY": 0,
                                },
                              ],
                            }
                          }
                        >
                          <View>
                            <View
                              style={
                                Object {
                                  "flex": 1,
                                  "margin": "50%",
                                }
                              }
                            >
                              <ActivityIndicator
                                animating={true}
                                color="#00a50b"
                                hidesWhenStopped={true}
                                size="large"
                              />
                            </View>
                            <RCTScrollView
                              refreshControl={
                                <RefreshControlMock
                                  onRefresh={[Function]}
                                  refreshing={true}
                                />
                              }
                            >
                              <RCTRefreshControl />
                              <View />
                            </RCTScrollView>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
              <View
                style={null}
                testID={undefined}
              />
              <View
                style={null}
                testID={undefined}
              />
            </View>
            <View
              collapsable={false}
            >
              <View
                collapsable={undefined}
                style={
                  Object {
                    "backgroundColor": "white",
                    "borderTopColor": "rgba(0, 0, 0, .2)",
                    "borderTopWidth": 0.5,
                    "flexDirection": "row",
                    "height": 49,
                  }
                }
              >
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  collapsable={undefined}
                  hitSlop={undefined}
                  nativeID={undefined}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "stretch",
                      "backgroundColor": "rgba(0, 0, 0, 0)",
                      "flex": 1,
                      "justifyContent": "flex-end",
                    }
                  }
                  testID={undefined}
                >
                  <View
                    style={
                      Object {
                        "flexGrow": 1,
                      }
                    }
                  >
                    <View
                      collapsable={undefined}
                      style={
                        Object {
                          "alignItems": "center",
                          "bottom": 0,
                          "justifyContent": "center",
                          "left": 0,
                          "opacity": 1,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                        }
                      }
                    />
                    <View
                      collapsable={undefined}
                      style={
                        Object {
                          "alignItems": "center",
                          "bottom": 0,
                          "justifyContent": "center",
                          "left": 0,
                          "opacity": 0,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                        }
                      }
                    />
                  </View>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    collapsable={undefined}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "backgroundColor": "transparent",
                        "color": "rgba(0, 0, 0, 1)",
                        "fontSize": 10,
                        "marginBottom": 1.5,
                        "textAlign": "center",
                      }
                    }
                  >
                    Próximos Rolês
                  </Text>
                </View>
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  collapsable={undefined}
                  hitSlop={undefined}
                  nativeID={undefined}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "stretch",
                      "backgroundColor": "rgba(0, 0, 0, 0)",
                      "flex": 1,
                      "justifyContent": "flex-end",
                    }
                  }
                  testID={undefined}
                >
                  <View
                    style={
                      Object {
                        "flexGrow": 1,
                      }
                    }
                  >
                    <View
                      collapsable={undefined}
                      style={
                        Object {
                          "alignItems": "center",
                          "bottom": 0,
                          "justifyContent": "center",
                          "left": 0,
                          "opacity": 0,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                        }
                      }
                    />
                    <View
                      collapsable={undefined}
                      style={
                        Object {
                          "alignItems": "center",
                          "bottom": 0,
                          "justifyContent": "center",
                          "left": 0,
                          "opacity": 1,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                        }
                      }
                    />
                  </View>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    collapsable={undefined}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "backgroundColor": "transparent",
                        "color": "rgba(90, 90, 90, 1)",
                        "fontSize": 10,
                        "marginBottom": 1.5,
                        "textAlign": "center",
                      }
                    }
                  >
                    Rolês passados
                  </Text>
                </View>
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  collapsable={undefined}
                  hitSlop={undefined}
                  nativeID={undefined}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "stretch",
                      "backgroundColor": "rgba(0, 0, 0, 0)",
                      "flex": 1,
                      "justifyContent": "flex-end",
                    }
                  }
                  testID={undefined}
                >
                  <View
                    style={
                      Object {
                        "flexGrow": 1,
                      }
                    }
                  >
                    <View
                      collapsable={undefined}
                      style={
                        Object {
                          "alignItems": "center",
                          "bottom": 0,
                          "justifyContent": "center",
                          "left": 0,
                          "opacity": 0,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                        }
                      }
                    />
                    <View
                      collapsable={undefined}
                      style={
                        Object {
                          "alignItems": "center",
                          "bottom": 0,
                          "justifyContent": "center",
                          "left": 0,
                          "opacity": 1,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                        }
                      }
                    />
                  </View>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    collapsable={undefined}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "backgroundColor": "transparent",
                        "color": "rgba(90, 90, 90, 1)",
                        "fontSize": 10,
                        "marginBottom": 1.5,
                        "textAlign": "center",
                      }
                    }
                  >
                    Cadastrar Rolê
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
